File: common/src/main/java/dev/obscuria/fragmentum/core/v1/common/signal/BaseSignal2.java
package dev.obscuria.fragmentum.core.v1.common.signal;

import dev.obscuria.fragmentum.api.v1.common.signal.Signal2;
import org.jetbrains.annotations.ApiStatus;

@ApiStatus.Internal
public final class BaseSignal2<P1, P2> extends BaseSignal<Signal2.Listener<P1, P2>> implements Signal2<P1, P2>
{
    public void emit(P1 p1, P2 p2)
    {
        this.emit(listener -> listener.consume(p1, p2));
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/core/tools/signal/BaseSignal2.kt
package dev.obscuria.fragmentum.core.tools.signal

import dev.obscuria.fragmentum.api.tools.signal.Signal2

class BaseSignal2<P1, P2> : BaseSignal<Signal2.Listener<P1, P2>>(), Signal2<P1, P2>
{
    override fun emit(p1: P1, p2: P2) = emit { it.consume(p1, p2) }
}